### DISCUSSION TITLE
Peek at value

### DISCUSSION THREAD

**Discussion by jon49** [vanjs-org/van#282] [General]
I would like to peek at a state value without actually causing a DOM rerender. But I always want to have the latest value when the user interacts with the UI and the DOM actually needs to be updated. Is there already an existing way to do this? If not I could put in a feature request.

So, the example would be this website https://jon49.github.io/timer/van where I have a timer. If I start the timer and then update the time it shuts off the timer. It should keep going and the user can choose to stop the timer themselves.

Here's the code: https://github.com/jon49/timer/blob/1f2be3f01832e2f97e62d95bcb3be049b8bab3bf/src/app.ts#L134

An idea I had, if this feature doesn't already exist, is to have a prefix that you could use when accessing the values, e.g., `data.$myValue` and then the van library would know that I want to access `myValue` but I don't want it to affect any rendering.

**Comment by Tao-VanJS** [comment]
I think you should put the line:

```js
let val = secondsLeft.val
```

inside the clause of if statement of `if (state.val === "running")`. This way, the change to `secondsLeft` will only trigger the re-rendering of DOM when `state.val` is `"running"`.

**Reply by jon49** [reply]
That doesn't fix the issue, it still resets the DOM. I think because of [this line](https://github.com/jon49/timer/blob/1f2be3f01832e2f97e62d95bcb3be049b8bab3bf/src/app.ts#L320). That code is called when the start button is pressed and then causes the timer to start clicking down to 0.

So, it still seems like I need to be able to peek at the values as I use them only once. I'll try and think of a different solution/work around.

**Comment by Tao-VanJS** [comment]
Ah..., I think I know what you're asking...

Basically you want the binding function inside `vanX.list` not being re-executed whenever there is an update to the fields of `timer`. If that is what you want, as a workaround, you can add the following lines at the beginning of the binding function:

```js
Object.assign(timer, timer)
```

This is because in VanJS [`1.3.0`](https://github.com/vanjs-org/van/discussions/275), a state won't be considered as a dependency of the binding function if we set the value of the state inside the binding function.

**Comment by Tao-VanJS** [comment]
Peeking seems to be a useful feature anyway. I'm considering to add it in the next VanJS and VanX release.

**Comment by Tao-VanJS** [comment]
Hi @jon49,

With the release of VanJS [`1.5.0`](https://github.com/vanjs-org/van/discussions/290) and VanX 0.3.0. You can use the property `rawVal` of `State` objects or [`vanx.raw`](https://vanjs.org/x#get-the-raw-field-value) to "peek" the state value without registering the dependency.

Hope it can help your use case :-)

**Reply by jon49** [reply]
Oh, great! I'll check that and update my app to use those values. I have some time this morning to work on it so perfect timing!

**Reply by jon49** [reply]
It seems to be working fine now.
